File: lua/ht/plugins/coding/nabla.lua
return {
  {
    "jbyuki/nabla.nvim",
    lazy = true,
    ft = { "latex", "markdown" },
    config = function()
      local show_nabla = function()
        local nabla = require("nabla")
        nabla.popup {
          border = "solid",
        }
      end
      ---@param ev vim.AutocmdCallback.Event?
      local setup_keymap = function(ev)
        local buf = 0
        if ev then
          buf = ev.buf
        end

        vim.keymap.set("n", "<leader>pf", show_nabla, {
          desc = "Preview formula",
          buffer = buf,
        })
      end
      vim.api.nvim_create_autocmd("FileType", {
        pattern = {
          "latex",
          "markdown",
        },
        callback = setup_keymap,
      })
      local ft = vim.api.nvim_get_option_value("ft", {
        buf = 0,
      })
      if ft == "latex" or ft == "markdown" then
        setup_keymap()
      end
    end,
  },
}

File: src/conf/plugins/coding/nabla.ts
import { ActionBuilder, Plugin, PluginOpts } from "@core/model";

function showNabla() {
  luaRequire("nabla").popup({
    border: "solid",
  });
}

const spec: PluginOpts<["nabla.show-nabla"]> = {
  shortUrl: "jbyuki/nabla.nvim",
  lazy: {
    ft: ["latex", "markdown"],
    keys: [
      {
        [1]: "<leader>pf",
        [2]: showNabla,
        ft: ["latex", "markdown"],
      },
    ],
  },
  providedActions: [
    ActionBuilder.start()
      .id("nabla.show-nabla")
      .title("Show Nabla")
      .condition((buf) => {
        return buf.filetype === "latex" || buf.filetype === "markdown";
      })
      .callback(() => {
        luaRequire("nabla").popup({
          border: "solid",
        });
      })
      .from("nabla.nvim")
      .category("Nabla")
      .build(),
  ],
};

export const plugin = new Plugin(spec);
